sympy/codegen/ast.py

```diff
@@ -106,6 +106,28 @@ class Assignment(Relational):
     Assignment(A, Matrix([[x, y, z]]))
     >>> Assignment(A[0, 1], x)
     Assignment(A[0, 1], x)
+
+    >>> from sympy import ccode
+    >>> from sympy import fcode
+    >>> from sympy.tensor import Indexed
+    >>> from sympy.tensor import Idx
+
+
+    >>> n = symbols('n', integer=True)
+
+    >>> ccode(Assignment(Indexed("A", Idx('i', (0, n))), 0)) #case_1
+    'for (int i=0; i<n + 1; i++){\n   A[i] = 0;\n}'
+
+    >>> fcode(Assignment(Indexed("A", Idx('i', (0, n))), 0)) #case_2
+    '      do i = 1, n + 1\n         A(i) = 0\n      end do'
+
+    As shown in case_1, the C loop runs from [0, n], i.e. the upper limit is
```

Comment:
The explanation can be accompanied with the corresponding doctest for brevity.

Reply:
@sidhantnagpal  pls take a look now. I've added the changes you suggested.